diesel/src/migrations/mod.rs

```diff
@@ -127,6 +127,22 @@ pub fn mark_migrations_in_directory<Conn>(conn: &Conn, migrations_dir: &Path)
     Ok(migrations)
 }
 
+// Returns true if there are outstanding migrations in the migrations directory, otherwise
+// returns false. Returns an `Err` if there are problems with migration setup.
+///
+/// See the [module level documentation](index.html) for information on how migrations should be
+/// structured, and where Diesel will look for them by default.
+pub fn any_pending_migrations<Conn>(conn: &Conn) -> Result<bool, RunMigrationsError> where
+    Conn: MigrationConnection
+{
+    let migrations_dir = find_migrations_directory()?;
+    let all_migrations = migrations_in_directory(&migrations_dir)?;
+    let already_run = conn.previously_run_migration_versions()?;
+
+    Ok(all_migrations.into_iter()
+        .any(|m| { !already_run.contains(&m.version().to_string()) }))
```

Comment:
FYI, no need for the inner braces around the closure. I'd probably also split off the `Ok` wrapping, i.e., `let pending = all…; \n Ok(pending)`